habitat_baselines/common/tensor_dict.py

```diff
@@ -110,18 +117,37 @@ def set(
                         continue
 
                 v = value[k]
+                tgt = self[k]
 
                 if isinstance(v, (TensorDict, dict)):
-                    self[k].set(index, v, strict=strict)
+                    assert isinstance(tgt, TensorDict)
```

Comment:
Maybe you want an a typing.Mapping object instead here? this doesn't make sense since this will sttically error here if a dict is passed in i

Reply:
If it's not, it should be cast to a TensorDict it seems like.

Reply:
It has to be a `TensorDict` that way it has a `set` method.  You can pass a dict in to `set`, but `self` must be a TensorDict